unit-tests/log/common.py

```diff
@@ -0,0 +1,38 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+from rspy import log, test
+import pyrealsense2 as rs
+import os
+
+
+def log_all():
+    rs.log( rs.log_severity.debug, "debug message" )
+    rs.log( rs.log_severity.info, "info message" )
+    rs.log( rs.log_severity.warn, "warn message" )
+    rs.log( rs.log_severity.error, "error message" )
+    # NOTE: fatal messages will exit the process and so cannot be tested
+    #rs.log( rs.log_severity.fatal, "fatal message" )
+    rs.log( rs.log_severity.none, "no message" )  # ignored; no callback
+
+
+n_messages = 0
+def message_counter( severity, message ):
+    global n_messages
+    n_messages += 1
+    log.d( message.full() )
+    #
+    test.check_equal( str(message), message.raw() )
```

Comment:
What does `.raw()` and `.full()` means?

Reply:
`.raw()` is the message content (the actual string that was sent to the `rs.log()`), whereas `.full()` is the whole line as output to the screen, with the filename, line number, etc.

Reply:
Where is this implemented? I can't find it.

Reply:
In `pyrealsense2.cpp` and `rs.hpp`